Solutions/IoTOTThreatMonitoringwithDefenderforIoT/Analytic Rules/IoTDenialofService.yaml

```diff
@@ -13,12 +13,14 @@ queryPeriod: 5m
 triggerOperator: gt
 triggerThreshold: 0
 tactics:
-  - Impact
+  - InhibitResponseFunction
+relevantTechniques:
+  - T0814
 query: |
   SecurityAlert
-  | where ProductName == "Azure Security Center for IoT"
+  | where ProviderName == "IoTSecurity"
   | where AlertName == "Suspicion of Denial Of Service Attack"
-  | where TimeGenerated <= ProcessingEndTime + 60m
+  | where tostring(parse_json(ExtendedProperties).isNew) == "True"
```

Comment:
In all the queries, we are using parse_json multiple times. This would impact the performance of the query. Please modify the query to parse once and utilize the results multiple times.
Example:
SecurityAlert
  | where ProviderName == "IoTSecurity"
  | where AlertName == "Suspicion of Denial Of Service Attack"
  |  extend ExtendedProperties = parse_json(ExtendedProperties)
  | where tostring(ExtendedProperties.isNew) == "True"
  | extend DeviceId = tostring(ExtendedProperties.DeviceId), SourceDeviceAddress = tostring(ExtendedProperties.SourceDeviceAddress), DestDeviceAddress = tostring(ExtendedProperties.DestinationDeviceAddress), RemediationSteps = tostring(parse_json(RemediationSteps)[0]), Protocol = tostring(ExtendedProperties.Protocol), AlertManagementUri = tostring(ExtendedProperties.AlertManagementUri)

Reply:
@devikamehra done